pkg/exchange/okex/okexapi/client.go

```diff
@@ -60,34 +61,26 @@ const (
 )
 
 type RestClient struct {
-	BaseURL *url.URL
-
-	client *http.Client
+	requestgen.BaseAPIClient
 
 	Key, Secret, Passphrase string
-
-	TradeService      *TradeService
-	PublicDataService *PublicDataService
-	MarketDataService *MarketDataService
 }
 
-func NewClient() *RestClient {
+func NewClient() (*RestClient, error) {
 	u, err := url.Parse(RestBaseURL)
```

Comment:
if we really want to check this error, could we move this parse into the init func.

constructor design should avoid return error, or the error should be checked outside of the constructor

```
var parsedBaseURL *url.URL
func init() {
      parsedBaseURL, err := url.Parse(RestBaseURL)
      if err != nil {
              panic(err)
      }
}
```